### PR TITLE
Code overhaul

### PR DESCRIPTION
**Pull request by ofersadan85**
I did a complete re-write of this code, for fun, and to try and teach you some pointers. I hope you like it

### CODE REVIEW

File: nasapotd.py [hunk 1]
```diff
@@ -1,46 +1,39 @@
...
-
-print('All Images Downloaded')
-print('Press enter to exit')
-input()
\ No newline at end of file
+
+def get_image_pages():
+    url = 'https://www.nasa.gov/rss/dyn/lg_image_of_the_day.rss'
+    r = requests.get(url)
+    soup = BeautifulSoup(r.content, 'html.parser')
```
Comment (ofersadan85) [inline nasapotd.py]: We don't have to split lines, we can use BeautifulSoup right here

File: nasapotd.py [hunk 2]
```diff
@@ -1,46 +1,39 @@
...
+    results = []
+    for item in soup.findAll('item'):
+        image_url = item.find('enclosure')['url']
+        image_extension = os.path.splitext(image_url)[-1]
+        image_title = item.find('title').text.replace(':', ' -').replace('"', '')  # We replace problematic characters
+        results.append({'url': image_url, 'title': image_title + image_extension})
```
Comment (ofersadan85) [inline nasapotd.py]: We just make a list of these and return them (both the url and the title with an extension)

File: nasapotd.py [hunk 3]
```diff
@@ -1,46 +1,39 @@
...
+    return results
+
+
+def download_one_image(url, save_path):
+    r = requests.get(url)
+    with open(save_path, 'wb') as f:
```
Comment (ofersadan85) [inline nasapotd.py]: This is how you properly open a file while making sure it also closes

File: nasapotd.py [hunk 4]
```diff
@@ -1,46 +1,39 @@
...
+        f.write(r.content)
+
+
+def download_all_images(save_folder_path):
+    os.makedirs(save_folder_path, exist_ok=True)
```
Comment (ofersadan85) [inline nasapotd.py]: With `exist_ok=True` we don't have to first check if the folder exists or not

File: nasapotd.py [hunk 5]
```diff
@@ -1,46 +1,39 @@
...
+    images = get_image_pages()
+    for i, img in enumerate(images):
+        img_path = os.path.join(save_folder_path, img['title'])
+        if not os.path.isfile(img_path):  # So we don't download and overwrite the same thing twice, ever
+            print('Downloading image ' + str(i) + '/' + str(len(images)) + ': ' + img_path)
```
Comment (ofersadan85) [inline nasapotd.py]: It's nice to show progress so we know how long we have to wait

**Comment by therealhe1ko** [comment]
Thank you very much. I appreciate it!

**Pull request merged**